.circleci/config.yml

```diff
@@ -2,24 +2,60 @@ version: 2
+
             sudo -E apt-get -yq update >> ~/apt-get-update.log 2>&1
-            sudo -E apt-get -yq --no-install-suggests --no-install-recommends install clang-format-15
-            git diff -U0 --no-color HEAD^1 *.c *.cc *.cpp *.h *.hpp *.i *.ic *.ih | clang-format-diff-15 -style=file -p1 >_GIT_DIFF
+            sudo apt-get install -yq --no-install-suggests --allow-unauthenticated --no-install-recommends clang-${COMPILER_VERSION} clang-tidy-${COMPILER_VERSION} clang-format-${COMPILER_VERSION} clang-tools-${COMPILER_VERSION} libc++-${COMPILER_VERSION}-dev libc++abi-${COMPILER_VERSION}-dev libldap2-dev curl libcurl4-openssl-dev bison libudev-dev libkrb5-dev libreadline-dev zlib1g-dev liblz4-dev libedit-dev libevent-dev protobuf-compiler libprotobuf-dev libprotoc-dev libldap2-dev libsasl2-dev libsasl2-modules-gssapi-mit cmake libicu-dev libtirpc-dev
+
+            # Download boost only if BASE_BRANCH is "8.0"
+            if [ "$BASE_BRANCH" = "8.0" ]; then
+              echo "BASE_BRANCH is 8.0, downloading and extracting Boost."
+              wget --progress=dot:giga -P ${BOOST_DIR} "https://archives.boost.io/release/${BOOST_VERSION//_/.}/source/boost_${BOOST_VERSION}.tar.gz"
+              tar -xzf "${BOOST_DIR}/boost_${BOOST_VERSION}.tar.gz" -C "${BOOST_DIR}"
+            else
+              echo "BASE_BRANCH is not 8.0, skipping Boost download."
+            fi
+
+            # Prepare compile_commands.json
+            cd ~/project
+            cmake -B /home/circleci/debug-build -DCMAKE_BUILD_TYPE=Debug -DWITH_BOOST=${BOOST_DIR} -DWITH_SSL=system -DWITH_AUTHENTICATION_LDAP=ON -DWITH_ROCKSDB=ON -DCMAKE_C_COMPILER=clang-${COMPILER_VERSION} -DCMAKE_CXX_COMPILER=clang++-${COMPILER_VERSION} -DCMAKE_CXX_FLAGS="-stdlib=libc++" -DCMAKE_EXE_LINKER_FLAGS="-stdlib=libc++" -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DWITH_SYSTEM_LIBS=ON -DWITH_FIDO=bundled -DWITH_ZSTD=bundled -DWITH_LZ4=bundled -DWITH_PROTOBUF=bundled ~/project
+
+            git remote add target "https://github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+            git fetch --no-tags --no-recurse-submodules target $BASE_BRANCH
+
+            echo "Checking clang-format results"
+            git diff -U0 "$(git merge-base HEAD target/$BASE_BRANCH)" *.c *.cc *.cpp *.h *.hpp *.i *.ic *.ih | clang-format-diff-$COMPILER_VERSION -style=file -p1 >_GIT_DIFF || true
             if [ ! -s _GIT_DIFF ]; then
-               echo The last git commit is clang-formatted;
+              echo The last git commit is clang-formatted;
+            else
+              cat _GIT_DIFF;
+              false;
+            fi
+
+            echo "Checking clang-tidy static code analysis results"
+            git diff -U0 "$(git merge-base HEAD target/$BASE_BRANCH)" *.c *.cc *.cpp *.h *.hpp *.i *.ic *.ih | clang-tidy-diff-${COMPILER_VERSION}.py -clang-tidy-binary clang-tidy-${COMPILER_VERSION} -p1 -path /home/circleci/debug-build -export-fixes=fixes.yaml -extra-arg=-stdlib=libc++ || true
```

Comment:
Do we really need `-extra-arg=-stdlib=libc++`? This flag should be included inside `CMAKE_EXPORT_COMPILE_COMMANDS`-generated commands file.

Reply:
Hmm, this is strange. So, unless this is a bug in clang-tidy itself (which is unlikely), it look like our server CMakeLists.txt file does not properly respect setting custom `CMAKE_CXX_FLAGS`.
Could you try setting `CMAKE_CXX_FLAGS_INIT` to "-stdlib=libc++" instead?
BTW, along with `CMAKE_EXE_LINKER_FLAGS` there are also
* `CMAKE_SHARED_LINKER_FLAGS`
* `CMAKE_STATIC_LINKER_FLAGS`

Reply:
You're right, Yura. I double checked it and CMAKE_CXX_FLAGS were propagated properly when set through cmake during compilation and we don't need to set the **_-extra-arg=-stdlib=libc++_** if it is already set at the cmake stage. Will remove it.